src/QMCHamiltonians/GridExternalPotential.h

```diff
@@ -0,0 +1,77 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2016 Jeongnim Kim and QMCPACK developers.
+//
+// File developed by: Jaron T. Krogel, krogeljt@ornl.gov, Oak Ridge National Laboratory
+//                    Mark A. Berrill, berrillma@ornl.gov, Oak Ridge National Laboratory
+//
+// File created by: Jaron T. Krogel, krogeljt@ornl.gov, Oak Ridge National Laboratory
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+#ifndef QMCPLUSPLUS_GRID_EXTERNAL_POTENTIAL_H
+#define QMCPLUSPLUS_GRID_EXTERNAL_POTENTIAL_H
+
+#include <QMCHamiltonians/QMCHamiltonianBase.h>
+#include <einspline/bspline.h>
+
+
+namespace qmcplusplus
+{
+struct GridExternalPotential : public QMCHamiltonianBase
+{
+  const ParticleSet& Ps;
+
+  UBspline_3d_d *spline_data;
```

Comment:
I guess spline_data is shared by all the threads and the evaluation functions are thread safe. So let us use shared pointer instead of raw pointer here.

Reply:
How would a shared pointer be implemented here?

Reply:
I mean using `std::shared_ptr<UBspline_3d_d> spline_data;`

Reply:
I tried changing:
```
UBspline_3d_d *spline_data;
```
to 
```
std::shared_ptr<UBspline_3d_d> spline_data;
```
but now I get a compilation error:
```
/home/kryczko/git/qmcpack/src/QMCHamiltonians/GridExternalPotential.cpp:120:61: error: cannot convert ‘std::shared_ptr<UBspline_3d_d>’ to ‘UBspline_3d_d*’ for argument ‘1’ to ‘void eval_UBspline_3d_d(UBspline_3d_d*, double, double, double, double*)’
       eval_UBspline_3d_d(spline_data, r[0], r[1], r[2], &val);
```

Reply:
`eval_UBspline_3d_d(spline_data.get(), r[0], r[1], r[2], &val);` will solve the problem.
https://en.cppreference.com/w/cpp/memory/shared_ptr

Reply:
Now I get the error:
```
/home/kryczko/git/qmcpack/src/QMCHamiltonians/GridExternalPotential.cpp: In member function ‘virtual bool qmcplusplus::GridExternalPotential::put(xmlNodePtr)’:
/home/kryczko/git/qmcpack/src/QMCHamiltonians/GridExternalPotential.cpp:87:61: error: no match for ‘operator=’ (operand types are ‘std::shared_ptr<UBspline_3d_d>’ and ‘UBspline_3d_d*’)
                                      BC, BC, BC, data.data());
```